capstone/src/main/webapp/scripts/poll.js

```diff
@@ -10,12 +11,39 @@ function getPollOptions() {
     .then(options => {
       const optionsContainer = document.getElementById("options-container");
       optionsContainer.innerHTML = "";
-      options.forEach(option => {
+      options["options"].forEach(option => {
         renderOptionElement(option);
       });
+      return options["votedOptions"];
+    })
+    .then(votedOptions => {
+      handleCheck(votedOptions);
     });
 }
 
+/**
+ * Handles whether checkbox is checked. Takes in list of ids of
+ * options for which current user has voted and checks if id of
+ * current checkbox is in that list.
+ * @param {array} votedOptions
+ */
+function handleCheck(votedOptions) {
+  const checkboxes = document.querySelectorAll("input[type=checkbox]");
+  for (let i = 0; i < checkboxes.length; i++) {
+    let checkbox = checkboxes[i];
+    if (
+      votedOptions.find(function(checkboxId) {
+        return checkbox.id == checkboxId;
+      })
+    ) {
+      checkbox.checked = true;
+    } else {
+      checkbox.checked = false;
+    }
+  }
+  return;
+}
```

Comment:
Could you walk me through here as to what is happening?

Reply:
I am going through the checkboxes to see which ones should be checked. votedOptions comes from the JSON which contains the ids of the checkboxes that the user has checked, so when I go through the list of checkboxes, I check to see which checkbox ids are in the votedOptions list so that I know which options to mark as checked

Reply:
Ah okay wonderful, so just to make sure I have it correct: for each checkbox you check if that checkbox exists by look through the entire list of voted options? Is there any way you can think of to make this more efficient?

Reply:
Oh yes! I couldn't get has() to work with sets but I was able to use the javascript version of hashmaps